megamek/src/megamek/server/Server.java

```diff
@@ -27109,23 +27109,86 @@ else if (secondaryEffects) {
                 js.setCICHits(js.getCICHits() + 1);
                 break;
             case Aero.CRIT_KF_DRIVE:
-                if (js == null) {
+                //Per SO construction rules, stations have no KF drive, therefore they can't take a hit to it...
+                if (js == null || js instanceof SpaceStation) {
                     break;
                 }
                 // KF Drive hit
-                r = new Report(9190);
-                r.subject = aero.getId();
-                reports.add(r);
-                js.setKFIntegrity(js.getKFIntegrity() - 1);
+                //Randomize the component struck - probabilities taken from the old BattleSpace record sheets
```

Comment:
The old BattleSpace rules are not the current rules and these changes are inconsistent with the official rules. It's fine to implement legacy rules but they need to be optional.

Reply:
Fixed.